src/AuthenticationInterface.php

```diff
@@ -15,10 +15,10 @@
 interface AuthenticationInterface
 {
     /**
-     * Authenticate the PSR-7 request and return a valid user
+     * Authenticate the PSR-7 request and return a valid identity
      * or null if not authenticated
      */
-    public function authenticate(ServerRequestInterface $request) : ?UserInterface;
+    public function authenticate(ServerRequestInterface $request) : ?IdentityInterface;
```

Comment:
**This is BC Break**, imho.

Anyone who has implement that interface need update code to change the RTH in implementation of that method.

Reply:
Thechnically speaking yes, but you need to change only the return type of `AuthentictionInterface::authenticate()` to `IdentityInterface`. Your existing code that returns a `UserInterface` object will remain as is. 

This code shows that is possible:
```php
declare(strict_types=1);

interface IdentityInterface {
    public function getIdentity(): string;
}

interface UserInterface extends IdentityInterface {
    public function getRoles() : iterable;
}

interface ClientInterface extends IdentityInterface {
    public function getScopes() : iterable;
}

interface AuthenticationInterface
{
    public function authenticate(string $param): ?IdentityInterface;
}

class Auth implements AuthenticationInterface
{
    public function authenticate(string $param): ?IdentityInterface {
        return new class implements UserInterface {
            public function getIdentity(): string {
                return 'identity';
            }
            public function getRoles() : iterable {
                return [];
            }
        };
    }
}

$a = new Auth(); // no error here
```

I think this can be an acceptable (small) BC breaks and we can document it in the release note.

Reply:
@ezimuel 

> Technically speaking yes, but you need to change only the return type

It is not my decision, but in my opinion we shouldn't release it in minor release. If someone was using it as it was it's going to stop working after running `composer update`.
For me the question is simply: _do we really follow semver or not?_

Reply:
@ezimuel 
> I think this can be an acceptable (small) BC breaks

"small" or "big" doesn't matter.

@webimpress 
> For me the question is simply: _do we really follow semver or not?_

We follow the semantic versioning!

Reply:
@ezimuel 
> …we can document it in the release note.

The goal should be: It must work without a user having to read the release notes or documentation.

Reply:
@froschdesign thanks for explain me how it works the versioning!!!

Reply:
@weierophinney it's up to you, for me we can release it in a minor/major, no matter. I was just trying to improve the existing code with minimum (one line change code) impact (btw, only if a user has implemented a specific interface). @webimpress and @froschdesign we don't leave in a perfect world.

Reply:
@froschdesign and @webimpress I removed the "without BC break" in the title and added BC break label.

Reply:
@ezimuel 
> thanks for explain me how it works the versioning!!!

I'm sorry, this is and was not my intention. I have only seen the outcry from users which run an update via Composer and did not read the release notes.

Sorry for the misunderstanding!

Reply:
@froschdesign it's ok. I would have to wait 10 minutes before answering (busy day). I see your point here and I actually agree, this is a BC break (even if it's a really small one).